test_reporting/junit_xml_parser.py

```diff
@@ -122,36 +128,91 @@ def validate_junit_xml_file(document_name):
     try:
         tree = ET.parse(document_name, forbid_dtd=True)
     except Exception as e:
-        raise JUnitXMLValidationError("could not parse provided XML document") from e
+        raise JUnitXMLValidationError(f"could not parse {document_name}: {e}") from e
 
     return _validate_junit_xml(tree.getroot())
 
 
+def validate_junit_xml_archive(directory_name):
+    if not os.path.exists(directory_name) or not os.path.isdir(directory_name):
+        raise JUnitXMLValidationError("file not found")
+
+    roots = []
+    metadata_source = None
+    metadata = {}
+    doc_list = glob.glob(os.path.join(directory_name, "*.xml"))
+    doc_list += glob.glob(os.path.join(directory_name, "**", "*.xml"))
+
+    total_size = 0
+    for document in doc_list:
+        total_size += os.path.getsize(document)
+
+    if total_size > MAXIMUM_XML_SIZE:
+        raise JUnitXMLValidationError("provided directory is too large")
```

Comment:
Total size is limited to 10MB, could this be too strict? The junit xml may contain captured log and have floated size.

Reply:
I think that's a pretty valid concern. I've bumped it up to 20MB which should be plenty. I think the largest individual file I've seen is ~250kb.